### PR TITLE
[UPDATE] more javax APIs migrated to jakarta

### PR DESCRIPTION
**Pull request by romainmoreau**
This PR migrates to the following Jakarta APIs:
- jakarta.annotation
- jakarta.inject
- jakarta.jms
- jakarta.persistence
- jakarta.servlet

It was started at the time of #2011 and it was recently unblocked with OpenJPA 4 release.

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]
Not too sure about the spring part but LGTM.

Thank you for this refactoring work!

File: server/apps/cassandra-app/src/assemble/license-for-binary.txt [hunk 1]
```diff
@@ -1103,37 +1103,3 @@ Within the lib directory
...
-            INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-            BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-            DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-            LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
-            OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
-            OF THE POSSIBILITY OF SUCH DAMAGE.
-
-        from JETM  http://jetm.void.fm
-            jetm-1.2.3.jar
-            jetm-optional-1.2.3.jar
```
Comment (chibenwa) [inline server/apps/cassandra-app/src/assemble/license-for-binary.txt:1139]: We are no longer using this, correct?

**Review by chibenwa — APPROVED** [review]
Masterpiece!

File: server/apps/cassandra-app/src/assemble/license-for-binary.txt (hunk 1 above)
Comment (romainmoreau) [reply server/apps/cassandra-app/src/assemble/license-for-binary.txt:1139]: My understanding is that it was removed a long time ago in 91848703279fa96cd1ab0b202b63554655e02fe7, and I suspect the files changed in this PR were overlooked.

**Pull request merged**